ArcGISRuntimeSDKQt_CppSamples/Analysis/DisplayUtilityAssociations/DisplayUtilityAssociations.cpp

```diff
@@ -0,0 +1,280 @@
+  {
+    if (!error.isEmpty())
+    {
+      qDebug() << error.message() << error.additionalMessage();
+      return;
+    }
+
+    m_mapView->setViewpoint(Viewpoint(Point(-9812698.37297436, 5131928.33743317, SpatialReference::webMercator()), targetScale));
+
+    // get all the edges and junctions in the network
+    QList<UtilityNetworkSource*> edges;
+    QList<UtilityNetworkSource*> junctions;
+
+    for (UtilityNetworkSource* networkSource : m_utilityNetwork->definition()->networkSources())
+    {
+      if (networkSource->sourceType() == UtilityNetworkSourceType::Edge)
+      {
+        edges.append(networkSource);
+      }
+      else if (networkSource->sourceType() == UtilityNetworkSourceType::Junction)
+      {
+        junctions.append(networkSource);
+      }
+    }
+
+    // add all edges that are not subnet lines to the map
+    for (UtilityNetworkSource* source : edges)
+    {
+      if (source->sourceUsageType() != UtilityNetworkSourceUsageType::SubnetLine && source->featureTable() != nullptr)
+      {
+        m_map->operationalLayers()->append(new FeatureLayer(source->featureTable(), this));
+      }
+    }
+
+    // add all junctions to the map
+    for (UtilityNetworkSource* source : junctions)
+    {
+      if (source->featureTable())
+      {
+        m_map->operationalLayers()->append(new FeatureLayer(source->featureTable(), this));
+      }
+    }
+
+    // create a renderer for the associations
+    UniqueValue* attachmentValue = new UniqueValue("Attachment", "", QVariantList{static_cast<int>(UtilityAssociationType::Attachment)}, m_attachmentSymbol, this);
+    UniqueValue* connectivityValue = new UniqueValue("Connectivity", "", QVariantList{static_cast<int>(UtilityAssociationType::Connectivity)}, m_connectivitySymbol, this);
+    UniqueValueRenderer* uniqueValueRenderer = new UniqueValueRenderer("", nullptr, QStringList{"AssociationType"}, QList<UniqueValue*>{attachmentValue, connectivityValue}, this);
+    m_associationsOverlay->setRenderer(uniqueValueRenderer);
+
+    // populate the legend
+    m_attachmentSymbol->createSwatch();
+    m_connectivitySymbol->createSwatch();
+  });
+
+  connect(m_utilityNetwork, &UtilityNetwork::associationsCompleted, this, [this](QUuid, const QList<UtilityAssociation*>& associations)
+  {
+    bool matchedGraphic = false;
+    for (UtilityAssociation* association : associations)
+    {
+      // check if the graphics overlay already contains the association
+      for (Graphic* graphic : *m_associationsOverlay->graphics())
+      {
+        if (graphic->attributes()->containsAttribute("GlobalId") && qvariant_cast<QUuid>(graphic->attributes()->operator[]("GlobalId")) == association->globalId())
+        {
+          matchedGraphic = true;
+          break;
+        }
+      }
+
+      // if the association is already in the overlay, don't add a new graphic
+      if (matchedGraphic)
+      {
+        matchedGraphic = false;
```

Comment:
maybe it's better to move `matchedGraphic` into the for loop scope of outer for loop and redeclare as `false` each time around. That way you don't have to worry about re-setting it